src/amberscript/parser.cc

```diff
@@ -2896,5 +2921,24 @@ Result Parser::ParseTolerances(std::vector<Probe::Tolerance>* tolerances) {
   return {};
 }
 
+Result Parser::ParseVirtualFile() {
+  auto token = tokenizer_->NextToken();
+  if (!token->IsIdentifier() && !token->IsString())
+    return Result("invalid virtual file path");
+  auto path = token->AsString();
+
+  auto r = ValidateEndOfStatement("VIRTUAL_FILE command");
+  if (!r.IsSuccess())
+    return r;
+
+  auto data = tokenizer_->ExtractToNext("END");
+
+  token = tokenizer_->NextToken();
+  if (!token->IsIdentifier() || token->AsString() != "END")
+    return Result("VIRTUAL_FILE missing END command");
+
+  return script_->GetVirtualFiles()->Add(path, data);
```

Comment:
`script_->SetVirtualFile(path, data)`

Reply:
See above. I could add helpers to the script to do forward on to the `VirtualFileStore` if you prefer?

Reply:
Yea, that's what I was thinking.

Reply:
Done.